src/library_wasmfs.js

```diff
@@ -85,6 +86,37 @@ var WasmfsLibrary = {
       // For file preloading, cwd should be '/' to begin with.
       return '/';
     }
+  },
+  _emscripten_write_js_file: function(index, buffer, length, offset) {
+    var size = wasmFS$JSMemoryFiles[index].length;
+    // Resize the typed array if the length of the write buffer exceeds its capacity.
+    if (offset + length >= size) {
+      var oldContents =wasmFS$JSMemoryFiles[index];
+      wasmFS$JSMemoryFiles[index] = new Uint8Array(offset + length);
+      wasmFS$JSMemoryFiles[index].set(oldContents.subarray(0, size));
```

Comment:
```suggestion
      var newContents = new Uint8Array(offset + length);
      newContents.set(oldContents);
      wasmFS$JSMemoryFiles[index] = newContents;
```
Avoid writing the large indexing expression twice (possible source of bugs).
Also avoid creating a subarray needlessly.